src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs

```diff
@@ -6,24 +6,53 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <summary>
 /// A resource that represents a MongoDB container.
 /// </summary>
-/// <param name="name">The name of the resource.</param>
-public class MongoDBServerResource(string name) : ContainerResource(name), IResourceWithConnectionString
+public class MongoDBServerResource : ContainerResource, IResourceWithConnectionString
```

Comment:
This is a breaking API change, we need to preserve the existing overload that takes a name.

Reply:
Is it better to overload the constructor or add new parameters with default to null?